CRM/ACL/BAO/ACL.php

```diff
@@ -525,12 +525,23 @@ private static function getGroupClause(array $groupIDs, string $operation): stri
     }
 
     if ($groupIDs) {
-      return "(
-        `contact_a`.id $operation (
-         SELECT contact_id FROM civicrm_group_contact WHERE group_id IN (" . implode(', ', $foundGroupIDs) . ") AND status = 'Added'
-           $groupContactCacheClause
-         )
-      )";
+      if ($foundGroupIDs) {
+        return "(
+          `contact_a`.id $operation (
+           SELECT contact_id FROM civicrm_group_contact WHERE group_id IN (" . implode(', ', $foundGroupIDs) . ") AND status = 'Added'
+             $groupContactCacheClause
+           )
+        )";
+      }
+      else {
+        // Edge case avoiding SQL syntax error if no $foundGroupIDs
+        return "(
+          `contact_a`.id $operation (
+           SELECT contact_id FROM civicrm_group_contact WHERE FALSE
+             $groupContactCacheClause
```

Comment:
This variable is initialized as `FALSE` (that's wrong btw it should be initialized as `''`) and will never contain a value here, so can be ommitted from this query. For that matter the entire subquery isn't needed. Probably this could just be 
```
return "(`contact_a`.id $operation (0) )";
```